sdk/src/main/java/com/hcaptcha/sdk/HCaptchaConfig.java

```diff
@@ -110,10 +110,26 @@ public class HCaptchaConfig implements Serializable {
 
     /**
      * Reset hCaptcha on timeout
+     * @deprecated use {@link #retryPredicate} to implement desired retry logic
      */
     @Builder.Default
+    @Deprecated
     private Boolean resetOnTimeout = false;
 
+    /**
+     * The lambda will decide should we retry or not
+     */
+    @Builder.Default
+    private IHCaptchaRetryPredicate retryPredicate = new IHCaptchaRetryPredicate() {
+        @Override
+        public boolean shouldRetry(HCaptchaConfig config, com.hcaptcha.sdk.HCaptchaException exception) {
+            if (Boolean.TRUE.equals(config.resetOnTimeout)) {
+                return exception.getHCaptchaError() == HCaptchaError.SESSION_TIMEOUT;
+            }
+            return false;
+        }
+    };
```

Comment:
We can still keep here a lambda instead an anonymous class declaration

```suggestion
    private IHCaptchaRetryPredicate retryPredicate = (HCaptchaConfig config, HCaptchaException exception) -> {
        if (Boolean.TRUE.equals(config.resetOnTimeout)) {
            return exception.getHCaptchaError() == HCaptchaError.SESSION_TIMEOUT;
        }
        return false;
    };
```

Reply:
Pushed https://github.com/hCaptcha/hcaptcha-android-sdk/pull/87/commits/7a5522169c975406a2a84c63b754b284fc348602